src/main/java/arona/task/Deadline.java

```diff
@@ -0,0 +1,49 @@
+package arona.task;
+
+import arona.exception.AronaInvalidDateException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+/**
+ * Represents a "Deadline" task,
+ * which is a task with a deadline time attached.
+ *
+ * @author Maximilliano Utomo
+ */
+public class Deadline extends Task {
+    /**
+     * The due date of the task.
+     */
+    protected LocalDate deadline;
+
+    /**
+     * A public constructor for the task.Deadline.
+     * @param desc - the description of the task
+     * @param deadline - the deadline of the task
+     */
+    public Deadline(String desc, String deadline) throws AronaInvalidDateException {
+        super(desc);
+        try {
+            this.deadline = LocalDate.parse(deadline);
+        } catch (DateTimeParseException e) {
+            throw new AronaInvalidDateException("");
+        }
+    }
+
```

Comment:
For consistency, you may consider adding a JavaDoc for the toDataFormat() function :)